app/main/views/frameworks.py

```diff
@@ -212,32 +212,40 @@ def get_framework_suppliers(framework_slug):
         Framework.slug == framework_slug
     ).first_or_404()
 
-    # TODO: Work out a way to order by current_framework_agreement.signed_agreement_returned_at
-    # Listing agreements is something done for Admin only (suppliers only retrieve their individual agreements)
-    # CCS always want to work from the oldest returned date to newest, so order by ascending date
+    # we're going to need an alias for the current_framework_agreement join
```

Comment:
Maybe I can ask you to explain this one in person tomorrow - there is much to SQLAlchemy to learn, and I might as well start with the hard bits :-)

(Where I'd got to in my rtfm'ing... )

- http://docs.sqlalchemy.org/en/latest/orm/loading_relationships.html#using-contains-eager-to-load-a-custom-filtered-collection-result doesn't seem to mention needing the aliases thing
- I don't really get why you need contains_eager for this anyway, if we're just loading a single related entity then can't we use joinedload?